packages/html-data/bin/crawler.ts

```diff
@@ -0,0 +1,53 @@
+import { Parser, defaultTreeAdapter, type DefaultTreeAdapterMap } from "parse5";
+
+type Document = DefaultTreeAdapterMap["document"];
+
+type ChildNode = DefaultTreeAdapterMap["childNode"];
+
+type Node = DefaultTreeAdapterMap["node"];
+
+type NodeWithChildren = Extract<ChildNode, { childNodes: ChildNode[] }>;
+
+type Element = DefaultTreeAdapterMap["element"];
+
+type Attribute = Element["attrs"][number];
+
+export const findTags = (
+  node: undefined | Node,
+  tagName: string,
+  result: NodeWithChildren[] = []
+): NodeWithChildren[] => {
+  if (node && "childNodes" in node) {
+    if ("tagName" in node && node.tagName === tagName) {
+      result.push(node);
+    }
+    for (const child of node.childNodes) {
+      findTags(child, tagName, result);
+    }
+  }
+  return result;
+};
+
+export const getAttr = (
+  node: NodeWithChildren,
+  name: string
+): undefined | Attribute => {
+  return node.attrs.find((attr) => attr.name === name);
```

Comment:
why the first attr and not the last one for example? is it obvious?

Reply:
Do you expect multiple attributes to have the same name?

Reply:
generally possible, but not sure how relevant here

Reply:
🤷